### PR TITLE
HYO-11 Buffer and Material Rework

### PR DESCRIPTION
**Pull request by christopherliu830**

### CODE REVIEW

File: src/graphics/buffers.odin [hunk 1]
```diff
@@ -71,10 +93,20 @@ buffers_shutdown :: proc() {
 /**
 * Create a buffer.
 */
-buffers_create :: proc(size: int, type: BufferType) -> Buffer {
+buffers_create :: proc(size: int, type: BufferType, alignment: int = mem.DEFAULT_ALIGNMENT) -> Buffer {
```
Comment (christopherliu830) [inline src/graphics/buffers.odin:96]: Forgot to change this.

File: src/graphics/buffers.odin [hunk 2]
```diff
@@ -71,10 +93,20 @@ buffers_shutdown :: proc() {
 /**
 * Create a buffer.
 */
-buffers_create :: proc(size: int, type: BufferType) -> Buffer {
+buffers_create :: proc(size: int, type: BufferType, alignment: int = mem.DEFAULT_ALIGNMENT) -> Buffer {
     return buffers_create_by_flags(size, buffers_default_flags(type), mem.DEFAULT_ALIGNMENT)
```
Comment (christopherliu830) [inline src/graphics/buffers.odin]: Forgot to change this to `alignment`.

File: src/graphics/descriptors.odin [hunk 3]
```diff
@@ -33,6 +33,7 @@ descriptors_create_pool :: proc (device: vk.Device, num_sets: int) -> vk.Descrip
     return builders.create_descriptor_pool(device, num_sets, pool_sizes[:]) 
 }
 
-
+descriptors_point :: proc() {
```
Comment (christopherliu830) [inline src/graphics/descriptors.odin]: Delete this

File: src/graphics/mats.odin [hunk 4]
```diff
@@ -69,52 +71,51 @@ mats_get_mat :: proc(cache: ^MaterialCache, name: string) -> ^Material {
     return &cache.materials[name]
 }
 
-mats_create_shader_effect :: proc(cache:            ^MaterialCache,
+mats_create_shader_effect :: proc(ctx:              ^RenderContext,
```
Comment (cs-fangyujen) [inline src/graphics/mats.odin]: Probably should keep render_pass as a parameter for different rendering types

Comment (christopherliu830) [reply src/graphics/mats.odin]: Ok

**Pull request merged**